tests/test_check_notebook.py

```diff
@@ -37,18 +39,87 @@ def test_check_notebook_remove_empty_cells(
     )
 
 
-@pytest.mark.parametrize("preserve_cell_metadata", [True, False])
+@pytest.mark.parametrize(
+    "preserve_cell_metadata",
+    [
+        [],
+        ["tags"],
+        ["other"],
+        ["tags", "special"],
+        ["nbformat", "tags", "special"],
+        None,
+    ],
+)
 def test_check_notebook_preserve_metadata(
     clean_notebook_with_metadata: nbformat.NotebookNode,
-    preserve_cell_metadata: bool,
+    preserve_cell_metadata: Union[Collection[str], None],
 ) -> None:
     """Test nb_clean.check_notebook when preserving cell metadata."""
-    assert (
-        nb_clean.check_notebook(
-            clean_notebook_with_metadata,
-            preserve_cell_metadata=preserve_cell_metadata,
+    assert nb_clean.check_notebook(
+        clean_notebook_with_metadata,
+        preserve_cell_metadata=preserve_cell_metadata,
+    ) is (
+        (preserve_cell_metadata is not None)
+        and (
+            preserve_cell_metadata == []
+            or {"tags", "special", "nbclean"}.issubset(preserve_cell_metadata)
+        )
+    )
```

Comment:
Given this is quite a long assertion now I think it'd be worth refactoring out `output` and `expected` variables:
```python
expected = (preserve_cell_metadata is not None...
output = nb_clean.check_notebook...
assert output is expected
```

Reply:
ok, no problem